src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -4802,86 +4798,22 @@ int wm_vuldet_collect_agent_software(scan_agent *agent, sqlite3 *db, scan_ctx_t*
         goto end;
     }
 
-    // Packages
-    if (OS_INVALID == wm_vuldet_get_last_software_scan(agent->agent_id, scan_id)) {
-        goto end;
-    }
-
-    const char* request_query = scan_ctx->scan_type == VU_PARTIAL_SCAN ? vu_queries[VU_SOFTWARE_REQUEST] : vu_queries[VU_SOFTWARE_FULL_REQ];
-
-    unsigned int i = 0;
-    snprintf(buffer, OS_SIZE_6144, request_query, agent->agent_id, scan_id, VU_MAX_PACK_REQ, i);
-    if (wm_vuldet_send_wdb(buffer)) {
-        mterror(WM_VULNDETECTOR_LOGTAG, VU_SOFTWARE_REQUEST_ERROR, scan_ctx->agent_id);
-        goto end;
-    }
-
-    while (size = wm_vuldet_recv_wdb(buffer, OS_SIZE_6144), size != OS_INVALID) {
-        if (size > 0) {
-            if (size < 10) {
-                break;
-            }
-            if (wm_vuldet_wdb_valid_answ(buffer)) {
-                buffer[0] = buffer[1] = ' ';
-                snprintf(json_str, OS_SIZE_6144 + 10, "{\"data\":%s}", buffer);
-            } else {
-                goto end;
-            }
-            if (obj) {
-                cJSON *new_obj;
-                cJSON *data;
-                if (new_obj = cJSON_ParseWithOpts(json_str, &jsonErrPtr, 0), !new_obj) {
-                    goto end;
-                } else if (!cJSON_IsObject(new_obj)) {
-                    free(new_obj);
-                    goto end;
-                }
-                data = cJSON_GetObjectItem(new_obj, "data");
-                if (data) {
-                    cJSON_AddItemToArray(package_list, data->child);
-                    free(data->string);
-                    free(data);
-                }
-                free(new_obj);
-            } else if (obj = cJSON_ParseWithOpts(json_str, &jsonErrPtr, 0), obj && cJSON_IsObject(obj)) {
-                package_list = cJSON_GetObjectItem(obj, "data");
-                if (!package_list) {
-                    goto end;
-                }
-            } else {
-                goto end;
-            }
-
-            i += VU_MAX_PACK_REQ;
-            snprintf(buffer, OS_SIZE_6144, request_query, agent->agent_id, scan_id, VU_MAX_PACK_REQ, i);
-            if (wm_vuldet_send_wdb(buffer)) {
-                mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_SOFTWARE_REQUEST_ERROR, scan_ctx->agent_id);
-                goto end;
-            }
-        } else {
-            goto end;
-        }
-    }
-
-    // Avoid checking the same packages again
-    snprintf(buffer, OS_SIZE_6144, vu_queries[VU_SYSC_UPDATE_SCAN], agent->agent_id, scan_id);
-    if (wm_vuldet_wdb_request(buffer, OS_SIZE_6144)) {
-        mtdebug1(WM_VULNDETECTOR_LOGTAG, VU_SOFTWARE_REQUEST_ERROR, scan_ctx->agent_id);
+    if (OS_INVALID == wm_vuldet_get_software(atoi(agent->agent_id), VU_PARTIAL_SCAN ? true : false, &package_list)) {
```

Comment:
Maybe, we should make this comparison in the ternary operator 
`scan_ctx->scan_type == VU_PARTIAL_SCAN`

Reply:
Fixed